src/sentry/integrations/vercel/integration.py

```diff
@@ -165,38 +165,58 @@ def update_organization_config(self, data):
             if mapping in old_mappings:
                 continue
             [sentry_project_id, vercel_project_id] = mapping
-
             sentry_project = Project.objects.get(id=sentry_project_id)
             enabled_dsn = ProjectKey.get_default(project=sentry_project)
             if not enabled_dsn:
                 raise IntegrationError("You must have an enabled DSN to continue!")
+            source_code_provider = self.get_source_code_provider(vercel_client, vercel_project_id)
+            if not source_code_provider:
+                raise IntegrationError(
+                    "You must connect your Vercel project to a Git repository to continue!"
+                )
             sentry_project_dsn = enabled_dsn.get_dsn(public=True)
+            uuid = uuid4().hex
 
-            org_secret = self.create_secret(
-                vercel_client, vercel_project_id, "SENTRY_ORG", sentry_project.organization.slug
+            sentry_auth_token = SentryAppInstallationForProvider.objects.get(
+                organization=sentry_project.organization.id, provider="vercel"
             )
-            project_secret = self.create_secret(
-                vercel_client,
-                vercel_project_id,
-                "SENTRY_PROJECT_%s" % sentry_project_id,
+            secret_names = [
+                "SENTRY_ORG_%s" % uuid,
+                "SENTRY_PROJECT_%s" % uuid,
+                "NEXT_PUBLIC_SENTRY_DSN_%s" % uuid,
+                "SENTRY_AUTH_TOKEN_%s" % uuid,
+            ]
+            values = [
+                sentry_project.organization.slug,
                 sentry_project.slug,
-            )
-            dsn_secret = self.create_secret(
-                vercel_client,
-                vercel_project_id,
-                "NEXT_PUBLIC_SENTRY_DSN_%s" % sentry_project_id,
                 sentry_project_dsn,
-            )
+                sentry_auth_token.sentry_app_installation.api_token.token,
```

Comment:
@ceorourke We actually shouldn't use the API token on the installation. There can actually be multiple on installation for internal integrations and this is effectively a deprecated way of doing it. Instead, we can use the logic that I had in vercel/webhook.py:

```
        installation_for_provider = SentryAppInstallationForProvider.objects.select_related(
            "sentry_app_installation"
        ).get(organization_id=organization_id, provider=self.provider)
        sentry_app_installation = installation_for_provider.sentry_app_installation

        # find a token associated with the installation so we can use it for authentication
        sentry_app_installation_token = (
            SentryAppInstallationToken.objects.select_related("api_token")
            .filter(sentry_app_installation=sentry_app_installation)
            .first()
        )
```

I would recommend putting this logic in a class method for `SentryAppInstallationForProvider`. See `SentryAppInstallationToken` for examples of how to do this in a class method.